frontend/src/metabase/static-viz/components/Gauge/GaugeContainer.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { Fragment } from "react";
+
+import { formatNumber } from "metabase/static-viz/lib/numbers";
+import { truncateText } from "metabase/static-viz/lib/text";
+import type { ColorGetter } from "metabase/static-viz/lib/colors";
+
+import {
+  START_ANGLE,
+  CHART_WIDTH,
+  GAUGE_OUTER_RADIUS,
+  CHART_VERTICAL_MARGIN,
+  GAUGE_INNER_RADIUS,
+  SEGMENT_LABEL_MARGIN,
+  DISTANCE_TO_MIDDLE_LABEL_ANCHOR,
+  MAX_SEGMENT_VALUE_WIDTH,
+  SEGMENT_LABEL_FONT_SIZE,
+} from "./constants";
+import {
+  removeDuplicateElements,
+  calculateRelativeValueAngle,
+  calculateSegmentLabelPosition,
+  calculateSegmentLabelTextAnchor,
+} from "./utils";
+
+import type { Card, Data, GaugeLabelData, Position } from "./types";
+import Gauge from "./Gauge";
+
+interface GaugeContainerProps {
+  card: Card;
+  data: Data;
+  getColor: ColorGetter;
+}
+
+export default function GaugeContainer({
+  card,
+  data,
+  getColor,
+}: GaugeContainerProps) {
+  const settings = card.visualization_settings;
+  const segments = settings["gauge.segments"];
+
+  const segmentMinValue = segments[0].min;
+  const segmentMaxValue = segments[segments.length - 1].max;
+
+  const center: Position = [
+    CHART_WIDTH / 2,
+    GAUGE_OUTER_RADIUS + CHART_VERTICAL_MARGIN,
+  ];
+
+  const value = data.rows[0][0];
+
+  const segmentMinMaxLabels: GaugeLabelData[] = segments
+    .flatMap(segmentDatum => {
+      return [segmentDatum.min, segmentDatum.max];
+    })
+    // gauge segments could be continuous i.e. the current max and the next min is the same value.
+    // So we should remove duplicate elements.
+    .reduce(removeDuplicateElements, [])
```

Comment:
I didn't test what would happen if we don't remove duplicate values, I guess the side effect is minimal, but I replicate the app viz behavior, seems like if the segments are continuous we don't render the same value at the same position twice.